contracts/libraries/DnGmxJuniorVaultManager.sol

```diff
@@ -682,11 +682,24 @@ library DnGmxJuniorVaultManager {
         // conversion of token into glp using batching manager
         // batching manager handles the conversion due to the cooldown
         // glp transferred to the vault on batch execution
-        uint256 glpReceived = state.batchingManager.depositToken(address(state.usdc), amount, usdgAmount);
+
+        uint256 glpReceived = _stakeGlp(state, address(state.usdc), amount, usdgAmount);
 
         emit GlpSwapped(glpReceived, amount, false);
     }
 
+    function _stakeGlp(
```

Comment:
not sure if it gets inline by the compiler, but since we know the token can be either WETH or USDC, should we use `mintAndStakeGlp` at specific locations and with particular tokens? and also approvals can be moved to grant allowances and allowance to batching manager should be removed